File: yht0827/2월/2월 5일/다음 큰 숫자.java
class Solution {
	public int solution(int n) {
		int answer = 0;
		String now = binary(n);
		int nowCount = 0;

		for (int i = 0; i < now.length(); i++) {
			if (now.charAt(i) == '1') {
				nowCount++;
			}
		}
		System.out.println(nowCount);

		while (true) {
			n++;
			int nextCount = 0;

			String next = binary(n);

			for (int i = 0; i < next.length(); i++) {
				if (next.charAt(i) == '1') {
					nextCount++;
				}
			}

			if (nowCount == nextCount) {
				break;
			}
		}

		return n;
	}

	public String binary(int n) {
		StringBuilder sb = new StringBuilder();
		while (n > 0) {
			sb.append(n % 2);
			n /= 2;
		}
		sb.reverse();

		return sb.toString();
	}
}

File: jeong-hyuk/2월5일/다음큰숫자.py
def count_one_bits(n):
    # 이진법으로 변환 후 1의 개수 반환
    return bin(n).count('1')

def solution(n):
    answer = n + 1
    
    # 주어진 숫자의 이진법 표현에서 1의 개수와 같은 수 중에서 가장 작은 수 찾기
    while count_one_bits(answer) != count_one_bits(n):
        answer += 1
    
    return answer